### ISSUE TITLE
Question: How to set up comparison rule <= (if possible)

### ISSUE THREAD

**Issue by ghost** [Seddryck/NBi#109]
Dear NBi-users

I just began working with this great add-on and read through the documention. I'm still a bit unsure wether or not it is possible so I hope someone can clarify.

I'm setting up a test-suite to Validate a SSAS cube against a relational data source, and compare 2 cubes (pre-prod/prod). It is expected that the cube should be larger or equal the one in production after burning the cube.

Is it possible to set up this rule?

My test is rather simple. Either mdx <-> mdx or mdx<-> sql

```
 <test name="description">
                    <system-under-test>
                        <execution>
                            <query file="....mdx" connectionString="@ref1" />
                        </execution>
                    </system-under-test>
                    <assert>
                        <equalTo>
                            <query file="....sql" connectionString="@ref2" />
                        </equalTo>
                    </assert>
                </test>
```

I tried specifying a `<column index="2" role="value" type="numeric" tolerance="+10%" />` but without luck, as it's treated like +/- 10%.

Thank you in advance

**Comment by FuegoArtificial** [comment]
I don't see exactly how, yet. But maybe there is an approach possible by using the [interval-feature](http://www.nbi.io/docs/compare-intervals/)?

Interesting however, we had the same question here as well, when just positive or negative tolerances are valid. In our case the opposite could not occur anyways so we dropped the question at this point...

**Comment by ghost** [comment]
I looked into the interval feature as well, but I didn't really understand how to utilize it in this scenario. Where does the code fit in in above situation? When <= should be applied to the whole column, rather than a single cell.

```
<row>
    <cell></cell>
    <cell>[100;+INF]</cell>
</row>
```

In case the above could work, the problem would be the requirement of a start-value. Unless this could be left out to use the actual value [ ; +INF]

**Comment by FuegoArtificial** [comment]
Do you want to compare cube sizes or measure values? What does "burning the cube" mean for you? Does the relational-DB vs. olap-DB test have the same requirement for tolerances like the test between the two cubes?
I'm hoping to get an idea about how to solve your issue by getting more context :)

**Comment by ghost** [comment]
I'm interested in comparing the measure-values, and the requirements are similar between the relational-DB and the two cubes. Measure values are allowed to be equal or greater.

By burning I meant Processing the cube :)

**Comment by FuegoArtificial** [comment]
Thanks!
What do you think about testing certain time frames for equality only, where you are sure that these values should not change, e.g. values from January, or dynamically last month's values?
It might not cover your requirement in total, but maybe it helps and it could be even a faster test.
If there might be slight differences between the two instances in January you might then apply the tolerance-property (positive and negative). :)
Basically, I'm thinking about testing the things which must be equal instead of different, achieving it by narrowing the data by querying certain time frames.

**Comment by ghost** [comment]
The thing is that its incremental, so the values must be equal or greater. I'm currently only looking at data from 2015, as this one is changing every time the cube is processed (assuming new data has been added to the relational DB).

All data prior to 2015 are static and no tolerance is allowed.

In terms of time it's not a factor, as it will be scheduled to run once every X days during night.

**Comment by FuegoArtificial** [comment]
Understood, so imagine you get data for August 2015 in the most recent processing. Would it be possible to test all data until July 2015 for equality? August will then be "more" in one instance implicitly?

**Comment by ghost** [comment]
The problem is that the data is usually quite stable until the end of the year. So I would like to test the whole year for =<, as lets say january could increase during may, and again during december.

**Comment by FuegoArtificial** [comment]
I see, then this approach is no option for you... :(
I don't have other ideas by now.

**Comment by ghost** [comment]
Thank you very much for your time and ideas though! I actually switched over from BI.Quality, as I could only specify > or < tolerances (adding + or - in front), and not =< and =>.

**Comment by Seddryck** [comment]
I'll check to implement this feature, sounds like a good extension.

**Comment by Seddryck** [comment]
I've built a preview for this feature at https://ci.appveyor.com/project/CdricLCharlier/nbi/build/1.11.0.99041

Just confirm if it fits your needs: `tolerance="+10%"` should do the job for the case you've described above.

**Comment by ghost** [comment]
The link seems unavailable. I'll check back later!

**Comment by Seddryck** [comment]
It sounds ok from this part of the world. :)

**Comment by ghost** [comment]
Found the issue. Company is blocking for some reason, but I found a work-around and tested the build. Seems to do exactly what I was looking for! Tried both "-10%" and "+10%".

**Comment by Seddryck** [comment]
The only remaining question is for negative values. When expecting -10 with a tolerance of +10% should we accept the interval [-10;-9] or [-11;-10] that's a tricky question. First solution is currently implemented.

**Comment by ghost** [comment]
I would think the first. As, the expected value (in my opinion) should be the standpoint and starting point.

Expected:
-10    tolerance = "+10%"  -> [-10;-9]
-10    tolerance = "-10%"  -> [-11;-10]
-10    tolerance = "10%"  -> [-11;-9]

But I guess it is how you look at it, and no "correct" answer exist?

**Comment by Seddryck** [comment]
Indeed, it's how I've implemented it. My self-argumentation was that if it's not a % tolerance but an absolute tolerance of "+1", I'd expect [-10;-9]. Following the same convention +10% becomes [-10;-9].
